Week 4/Contest (Apr 3)/Tajna/elijah.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+partial class Program
+{
+  static void Main()
+  {
+    string msg = Console.ReadLine();
+    int n = msg.Length;
+
+    int rows = -1;
+    int columns = -1;
+    for (int r = msg.Length; r > 1 && rows == -1 && columns == -1; r--)
+    {
+      for (int c = r; c < msg.Length; c++)
+      {
+        if (c * r == n)
+        {
+          rows = r;
+          columns = c;
+          break;
+        }
+      }
+    }
+
+    if (rows == -1 || columns == -1)
+    {
+      rows = 1;
+      columns = msg.Length;
+    }
+    else if (columns > rows)
```

Comment:
This bit shouldn't be necessary, but it was slightly faster than fixing my original calculation.